app_flutter/lib/service/appengine_cocoon.dart

```diff
@@ -46,6 +46,48 @@ class AppEngineCocoonService implements CocoonService {
     return _commitStatusesFromJson(jsonResponse['Statuses']);
   }
 
+  @override
+  Future<bool> fetchTreeBuildStatus() async {
+    /// This endpoint returns JSON {AnticipatedBuildStatus: [BuildStatus]}
+    http.Response response =
+        await _client.get('$_baseApiUrl/public/build-status');
+
+    if (response.statusCode != HttpStatus.ok) {
+      throw HttpException(
+          '$_baseApiUrl/public/build-status returned ${response.statusCode}');
+    }
+
+    Map<String, Object> jsonResponse = jsonDecode(response.body);
+
+    if (!_isBuildStatusResponseValid(jsonResponse)) {
```

Comment:
Better log this response, otherwise when we see the exception, we don't know why/how it's malformed.

Reply:
This will print the error to a local console. Should we think about implementing a logging system such as Firebase Crashlytics so users don't have to report these errors?

Reply:
The issue with logging to console is that the #1 usecase for this will be the TVs in the immediate aread.

Reply:
Not only the TV, I always have the build dashboard open on my chrome, so it would be nice to know the errors when they happen.

Reply:
When errors happen on the TV, we just need to know an error is occurring. We do not need to know what the error is. The only time we would want to know what the error is on the TV is when we can't reproduce it on any other device. However, in that case the issue is most likely with the TV and not the application.

Thus, showing a generic error UI would indicate that someone needs to check what is going on by opening the dashboard on their computer to investigate further.